File: pipelines/load_products_bronze/code/src/main/scala/graph/Target_1.scala
package graph

import io.prophecy.libs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import config.ConfigStore._

object Target_1 {

  def apply(spark: SparkSession, in: DataFrame): Unit = {
    Config.fabricName match {
      case "recipes_fabric" =>
        in.write
          .format("csv")
          .option("header", true)
          .option("sep",    ",")
          .mode("append")
          .save(
            "dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/products/bronze/"
          )
      case "anshuman2" =>
        import org.apache.avro.Schema
        var writer = in.write.format("avro")
        writer = writer
        writer.save(
          "dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/products/bronze"
        )
      case _ =>
        throw new Exception("No valid dataset present to read fabric")
    }
  }

}

File: pipelines/load_acc_status_bronze/code/job/graph/Target_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from job.config.ConfigStore import *
from job.udfs.UDFs import *

def Target_1(spark: SparkSession, in0: DataFrame):
    if Config.fabricName == "recipes_fabric":
        in0.write\
            .option("header", True)\
            .option("sep", ",")\
            .mode("append")\
            .option("separator", ",")\
            .option("header", True)\
            .csv("dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/acc_status/bronze/")
    elif Config.fabricName == "anshuman2":
        in0.write\
            .option("header", True)\
            .option("sep", ",")\
            .mode("overwrite")\
            .option("separator", ",")\
            .option("header", True)\
            .csv("dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/acc_status/bronze/")
    elif Config.fabricName == "anshuman":
        in0.write\
            .option("header", True)\
            .option("sep", ",")\
            .mode("overwrite")\
            .option("separator", ",")\
            .option("header", True)\
            .csv("dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/acc_status/bronze/")
    else:
        raise Exception("No valid dataset present to read fabric")
